### PR TITLE
Use full size for adaptive overlay

### LINKED ISSUE #138: Wrong size of adaptive icon overlay
At least on my device/launcher (One Plus 7T Pro), the generated adaptive icon has some issues. Due to the `72dp` size of the overlay, the overlay does not fill the whole foreground.

screenshot-1616851191457

### PR DESCRIPTION
**Pull request by brudaswen**
Rendering every overlay with 108dp and draw the actual content in the 72dp center. The background of the ribbon fills the whole image to ensure that potential foreground animations do not show the borders of the rectangles.

Fixes #138

### CODE REVIEW

**Comment by brudaswen** [comment]
@mateuszkwiecinski I updated the PR (still WIP): I now have a solution that produces the same results in the OnePlus Launcher and the Nove Launcher. I still have to adapt the label text size, since I deviated too much from the original version, but maybe you can already have a first look to give some early feedback?

**Comment by mateuszkwiecinski** [comment]
Hey @brudaswen 👋 Thank you for working on the issue 🙏 
I admit I haven't considered other launchers when picking values used to position overlays and most of the code positioning the label comes from the older plugin version. 

Could you confirm that the `One Plus Launcher` your're referring to is: https://play.google.com/store/apps/details?id=net.oneplus.launcher so I could verify that as well? 
I'll try to find some time later this week to dive into changes you suggested. (I'll definitelly be available during the weekend)

>Rendering every overlay with 108dp and draw the actual content in the 72dp center.

I just realized I misunderstood the documentation 🤦 It states _"Both layers must be sized at 108 x 108 dp."_, so definitiely current plugin behavior is wrong so the change you're suggesting is definitely a right one 👌 

>Introduced an adaptivePadding setting that allows to enable/disable padding for adaptive icons

My initial thought on that is AFAIU this won't be used too often and it is neede to preserve backwards compatibility, so if you manage to fix paddings and text sizes and to make it look closer to the current version I think the new api can be _hidden_ a bit more (i.e. moved to `drawingOptions` instead of adding new boolean parameter). I'd like to see the default plugin configuration to produce similar output as the current one.

If there still is be a slight differece between current version and what you come up with - we'll bump major version 🤷 Although, if I'm not mistaken, it should be _doable_ to make existing screenshot tests pass, even with these changes you suggested.

> Discuss:

As for now, I don't understand what both `Discuss:` are precisely about, I'll speak my mind after I'll refresh my memory on current plugin behavior.

btw. I know recording screenshots for the screenshot tests is a bit painful, so I can help you with that, if needed. Or I can provide some guidance. I commited `ini` file needed to create the same emulator and I tried to create `recordAll.sh` script but I haven't used it for a while so I'm not sure if it works. I'm researching using different [library](https://github.com/Karumi/Shot) which would allow me to simply create single gradle task 🤷

**Comment by brudaswen** [comment]
@mateuszkwiecinski A little Christmas present! 🎄

**Comment by mateuszkwiecinski** [comment]
Wow thanks! ❤️  I'll try to review and merge these changes somewhen this week. At the first glance all looks great 👌

File: sample/build.gradle [hunk 1]
```diff
@@ -76,26 +85,133 @@ subprojects {
...
         }
-        tasks.register("recordAll") {
+
+        tasks.register("installAll") {
+            group = 'install'
             android.applicationVariants.configureEach { variant ->
-                dependsOn("record${variant.flavorName.capitalize()}DebugAndroidTestScreenshotTest")
+                dependsOn("install${variant.name.capitalize()}")
+            }
+        }
```
Comment (mateuszkwiecinski) [inline sample/build.gradle:127]: 📝 
Comment to myself: `installAll` task doesn't work properly due to ignored `applicationIdSuffix` config 👀  
I'll investigate it and try to fix. I guess this has been introduced when the project started to be compiled agains AGP 7.x which made the `applicationVariants` deprecated 🤷

**Review by mateuszkwiecinski — APPROVED** [review]
Wow, that's all great 🎉 
I left couple of comments but I'll them myself as they are totally minor and they won't change anything in how the plugin works. As soon as I'm done I'll make a release :) 

Thank you for all the great work 🙏

**Pull request merged**